### PR TITLE
$postOptions

### PR DESCRIPTION
**Pull request by Herteby**
I've added support for limit, skip and sort. For sort I'm using Minimongo.Sorter.
I also renamed $postfilter to $postfilters, but kept $postfilter as an alias.

### CODE REVIEW

**Comment by Herteby** [comment]
At the moment this all happens on the client though. Would be good if it could be done on the server, though I imagine that would be somewhat complicated?

File: lib/query/lib/createGraph.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import ReducerNode from '../nodes/reducerNode.js';
 import dotize from './dotize.js';
 import createReducers from '../reducers/lib/createReducers';
 
-const specialFields = ['$filters', '$options', '$postFilter'];
+const specialFields = ['$filters', '$options', '$postFilters','$postOptions','$postFilter']; //keep $postFilter for legacy support
```
Comment (theodorDiaconu) [inline lib/query/lib/createGraph.js]: use space after comma

File: lib/query/lib/prepareForDelivery.js [hunk 2]
```diff
@@ -24,13 +26,31 @@ export default (node) => {
...
     node.collectionNodes.forEach(collectionNode => {
-        applyPostFilter(collectionNode);
+        applyPostFilters(collectionNode);
+    })
+}
+export function applyPostOptions(node) {
+    const options = node.props.$postOptions;
+    if (options) {
+        if(options.limit || options.skip){
+            const start = options.skip || 0, end = options.limit ? options.limit + start : -0;
```
Comment (theodorDiaconu) [inline lib/query/lib/prepareForDelivery.js]: declare variables each on their own line.
Also -0 ?

File: lib/query/lib/prepareForDelivery.js [hunk 3]
```diff
@@ -24,13 +26,31 @@ export default (node) => {
...
+            node.results = node.results.slice(start, end);
+        }
+        if(options.sort){
+            const sorter = new Minimongo.Sorter(options.sort);
```
Comment (theodorDiaconu) [inline lib/query/lib/prepareForDelivery.js]: nice

**Comment by theodorDiaconu** [comment]
very good overall, pay more attention to coding standards pls

File: lib/query/lib/prepareForDelivery.js (hunk 2 above)
Comment (Herteby) [inline lib/query/lib/prepareForDelivery.js]: I thought -0 would work, as negative numbers represent offset from the end of the array. But I tested it now and it doesn't actually work!
Changing it to node.results.length

File: lib/query/lib/prepareForDelivery.js [hunk 4]
```diff
@@ -24,13 +26,31 @@ export default (node) => {
...
 
     node.collectionNodes.forEach(collectionNode => {
-        applyPostFilter(collectionNode);
+        applyPostFilters(collectionNode);
+    })
+}
+export function applyPostOptions(node) {
+    const options = node.props.$postOptions;
+    if (options) {
+        if(options.limit || options.skip){
```
Comment (Herteby) [inline lib/query/lib/prepareForDelivery.js]: Yeah, sorry! :)
How about adding .eslintrc and/or .jsbeautifyrc files to the root of the project?
Btw, I normally indent using tabs (tab spacing 2) and [don't use semicolons](https://github.com/yyx990803/semi#why) :stuck_out_tongue:

Comment (theodorDiaconu) [inline lib/query/lib/prepareForDelivery.js]: Good idea. I usually prefer 4 because it forces you to decouple and avoid spaghetti

**Comment by Herteby** [comment]
Could be good to have a unit test to confirm that sort works as expected, though I'm not sure how to test that :thinking:
At least the current test makes sure that sort doesn't crash or return zero results.

**Comment by theodorDiaconu** [comment]
Sorter is already tested in their library, if we used the correct api it's enough. That was my thinking when i merged it.

**Pull request merged**